client/account/security/TwoFactorTOTP.js

```diff
@@ -0,0 +1,198 @@
+			if (!totpEnabled) { return false; }
+			const result = await checkCodesRemainingFn();
+			setCodesRemaining(result.remaining);
+		};
+		updateCodesRemaining();
+	}, [checkCodesRemainingFn, totpEnabled]);
+
+	const handleEnableTotp = useCallback(async () => {
+		try {
+			const result = await enableTotpFn();
+
+			setTotpSecret(result.secret);
+			setQrCode(qrcode(result.url, { size: 200 }));
+
+			setRegisteringTotp(true);
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error });
+		}
+	}, [dispatchToastMessage, enableTotpFn]);
+
+	const handleDisableTotp = useCallback(async () => {
+		const onDisable = async (authCode) => {
+			try {
+				const result = await disableTotpFn(authCode);
+
+				if (!result) {
+					return dispatchToastMessage({ type: 'error', message: t('Invalid_two_factor_code') });
+				}
+
+				dispatchToastMessage({ type: 'success', message: t('Two-factor_authentication_disabled') });
+			} catch (error) {
+				dispatchToastMessage({ type: 'error', message: error });
+			}
+		};
+
+		setModal(<VerifyCodeModal onVerify={onDisable} onCancel={closeModal}/>);
+	}, [closeModal, disableTotpFn, dispatchToastMessage, setModal, t]);
+
+	const handleVerifyCode = useCallback(async () => {
+		try {
+			const result = await verifyCodeFn(authCode);
+
+			if (!result) {
+				return dispatchToastMessage({ type: 'error', message: t('Invalid_two_factor_code') });
+			}
+			const codes = `<pre><code> ${ mapCodes(result.codes) }</pre></code>`;
+			setModal(<BackupCodesModal codes={codes} onClose={closeModal}/>);
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error });
+		}
+	}, [authCode, closeModal, dispatchToastMessage, setModal, t, verifyCodeFn]);
+
+	const handleRegenerateCodes = useCallback(() => {
+		const onRegenerate = async (authCode) => {
+			try {
+				const result = await regenerateCodesFn(authCode);
+
+				if (!result) {
+					return dispatchToastMessage({ type: 'error', message: t('Invalid_two_factor_code') });
+				}
+				const codes = `<pre><code> ${ mapCodes(result.codes) }</pre></code>`;
+				setModal(<BackupCodesModal codes={codes} onClose={closeModal}/>);
+			} catch (error) {
+				dispatchToastMessage({ type: 'error', message: error });
+			}
+		};
+
+		setModal(<VerifyCodeModal onVerify={onRegenerate} onCancel={closeModal}/>);
+	}, [closeModal, dispatchToastMessage, regenerateCodesFn, setModal, t]);
+
+	return <Box display='flex' flexDirection='column' alignItems='flex-start' {...props}>
+		<Margins blockEnd='x8'>
+			<Box fontScale='s2'>{t('Two-factor_authentication')}</Box>
+			{!totpEnabled && !registeringTotp && <>
+				<Box>{t('Two-factor_authentication_is_currently_disabled')}</Box>
+				<Button primary onClick={handleEnableTotp}>{t('Enable_two-factor_authentication')}</Button>
+			</>}
+			{!totpEnabled && registeringTotp && <>
+				<Box>{t('Scan_QR_code')}</Box>
+				<Box>{t('Scan_QR_code_alternative_s', { code: totpSecret })}</Box>
```

Comment:
Maybe it would be better to remove the code from the i18n string. I think there is something like a "copy this code" component somewhere else, right?